core.py

```diff
@@ -432,9 +432,13 @@ def photon_channels(self, points):
         # Randomly assign to channel given probability of each channel
         # Sum probabilities over channels should be 1
         self._photon_channels = np.array([])
+        channels = np.array(self.config['channels_in_detector']['tpc']).astype(int)
+
+        self._photon_channels = np.zeros(len(self._photon_timings), dtype='<i8')
+        count_start = 0
```

Comment:
1) Pulling channel out config actually takes time
2) np.append is really slow, as it reallocates memory